docs/topics/multiplatform-mobile/multiplatform-mobile-upgrade-app.md

```diff
@@ -0,0 +1,348 @@
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class RocketLaunch (
+    SerialName("flight_number")
+    val flightNumber: Int,
+    @SerialName("mission_name")
+    val missionName: String,
+    @SerialName("launch_date_utc")
+    val launchDateUTC: String,
+    @SerialName("launch_success")
+    val launchSuccess: Boolean?,
+)
+```
+
+* The `RocketLaunch` class is marked with the `@Serializable` annotation, so that the `kotlinx.serialization` plugin can
+automatically generate a default serializer for it.
+* The `@SerialName` annotation allows redefining field names, making it possible to declare properties in data classes
+with more readable names.
+
+### Connect HTTP client
+
+1. Create a Ktor `HTTPClient` instance to execute network requests and parse the resulting JSON:
+
+    ```kotlin
+    import io.ktor.client.*
+    import io.ktor.client.plugins.contentnegotiation.*
+    import io.ktor.serialization.kotlinx.json.*
+    import kotlinx.serialization.json.Json
+    
+    class Greeting {
+        private val httpClient = HttpClient {
+            install(ContentNegotiation) {
+                json(Json {
+                    prettyPrint = true
+                    isLenient = true
+                    ignoreUnknownKeys = true
+                })
+            }
+        }
+    }
+    ```
+
+    To deserialize the result of the GET request,
+    the [ContentNegotiation Ktor plugin](https://ktor.io/docs/serialization-client.html#register_json) and the JSON
+    serializer are used.
+
+2. In the `greeting()` function, retrieve the information about rocket launches by calling the `httpClient.get()`
+   method and find the latest one:
+
+    ```kotlin
+    class Greeting {
+        // ...
+        @Throws(Exception::class)
+        suspend fun greeting(): String {
+            val rockets: List<RocketLaunch> = httpClient.get("https://api.spacexdata.com/v3/launches").body()
+            val firstSuccessLaunch = rockets.first { it.launchSuccess == true }
+            return "Guess what it is! > ${Platform().platform.reversed()}!" +
+                "\nThere are only ${daysUntilNewYear()} left! 🎅🏼 " +
+                "\nThe first success launch was ${firstSuccessLaunch.launchDateUTC} 🚀"
+        }
+    }
+    ```
+
+    The `suspend` modifier in the `greeting()` function is necessary because it now contains a call to `get()`. It's a
+    a suspend function that has an asynchronous operation to retrieve data over the internet and can only be called from
+    within a coroutine or another suspend function. The network request will be executed in the HTTP client's thread pool.
+
+### Add internet access permission
+
+To access the internet, the Android application needs appropriate permission. Since all network requests are made from the
+shared module, it makes sense to add the internet access permission to its manifest.
+
+Update your `shared/src/androidMain/AndroidManifest.xml` file the following way:
```

Comment:
We should pay attention that the package should be the one used in the user's application because if I copy and paste the whole file text, I have 'com.jetbrains.onboarding' which is different from my package